### PR TITLE
[DUOS-753][risk=no] Add a spinner/loading icon over the sign in button while it is loading

### PR DESCRIPTION
**Pull request by lindojong**
Screen Shot 2020-10-07 at 12 46 46 PM

Users were confused by the greyed out log in button appearing before being usable. We had some spinner logic for when we fetch the client id on our side, but the default 3rd party login component had an additional loading step (greyed out button) that temporarily disables the button and prevents login. These changes capture when the component is disabled and displays the spinner instead until the login component is fully ready for use.

Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [x] PR is labeled with a Jira ticket number and includes a link to the ticket
- [x] PR is labeled with a security risk modifier [no, low, medium, high] 
- [x] PR describes scope of changes

In all cases:

- [ ] Get a minimum of one thumbs worth of review, preferably 2 if enough team members are available
- [ ] Get PO sign-off for all non-trivial UI or workflow changes
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/components/SignIn.js [hunk 1]
```diff
@@ -125,21 +125,36 @@ export const SignIn = hh(class SignIn extends Component {
...
+      });
+  };
+
   render() {
 
     let googleLoginButton;
 
     if (this.state.clientId === '') {
-      googleLoginButton = div({ style: { 'position': 'relative', 'marginTop': '20px', 'marginLeft': '45px', 'zIndex': '10000' } }, [
+      googleLoginButton = div({ style: { 'position': 'absolute', 'top': '45px', 'right': '45px', 'zIndex': '10000' } }, [
```
Comment (rushtong) [inline src/components/SignIn.js]: Suggestion. This aligns the spinner under the `Already Registered?` text and when the google button displays, that text jumps to the left. If we used this style: 
```
{ textAlign: 'center', height: '44px', width: '180px' }
```
Then the text stays in the same place and the spinner is replaced with the button in the exact same position.

**Review by rushtong — APPROVED** [review]
One suggestion inline, but looks good!

File: src/components/SignIn.js (hunk 1 above)
Comment (lindojong) [reply src/components/SignIn.js]: Yeah, I was going for more visual "clarity" by having the elements jump a bit as they reposition into place when the button loads, but I'm happy to realign the spinner!

Comment (rushtong) [reply src/components/SignIn.js]: Thank you for the changes!

**Pull request merged**